### PR TITLE
Euler new updates

### PR DESCRIPTION
**Pull request by shriyatyagii**
- Removed swap function
- Changed `balanceOf` -> `balanceOfUnderlying` in withdraw
- Added max check in `burn` 
image

### CODE REVIEW

File: contracts/mainnet/connectors/euler/main.sol [hunk 1]
```diff
@@ -376,105 +376,37 @@ abstract contract Euler is Helpers {
...
 
 		IEulerDToken dToken = IEulerDToken(markets.underlyingToDToken(_token));
 
-		dToken.approveDebt(subAccountId, debtReceiver, _amt);
+		dToken.approveDebt(subAccountId, debtSender, amt);
 
-		setUint(setId, _amt);
+		setUint(setId, amt);
 
 		_eventName = "LogApproveDebt(uint256,address,address,uint256)";
```
Comment (pradyuman-verma) [inline contracts/mainnet/connectors/euler/main.sol]: Should we update the event name as well `LogApproveSpenderDebt`?
```suggestion
		_eventName = "LogApproveDebt(uint256,address,address,uint256)";
```

Comment (shriyatyagii) [reply contracts/mainnet/connectors/euler/main.sol]: Can do! I've updated

**Review by pradyuman-verma — APPROVED** [review]

File: contracts/mainnet/connectors/euler/main.sol [hunk 2]
```diff
@@ -95,12 +95,14 @@ abstract contract Euler is Helpers {
 
 		TokenInterface tokenContract = TokenInterface(_token);
 		IEulerEToken eToken = IEulerEToken(markets.underlyingToEToken(_token));
-		_amt = _amt == uint256(-1) ?  eToken.balanceOf(address(this)) : _amt;
-		uint256 initialBal = tokenContract.balanceOf(address(this));
+
+		address _subAccount = getSubAccount(address(this), subAccount);
+		_amt = _amt == uint256(-1) ?  eToken.balanceOfUnderlying(_subAccount) : _amt;
+		uint256 initialBal = tokenContract.balanceOf(_subAccount);
```
Comment (thrilok209) [inline contracts/mainnet/connectors/euler/main.sol]: @shriyatyagii This should be `address(this)` only

Comment (thrilok209) [reply contracts/mainnet/connectors/euler/main.sol]: We are checking withdrawal amount to dsa. not to subaccount address

File: contracts/mainnet/connectors/euler/main.sol [hunk 3]
```diff
@@ -95,12 +95,14 @@ abstract contract Euler is Helpers {
...
 
 		eToken.withdraw(subAccount, _amt);
 
-		uint256 finalBal = tokenContract.balanceOf(address(this));
+		uint256 finalBal = tokenContract.balanceOf(_subAccount);
```
Comment (thrilok209) [inline contracts/mainnet/connectors/euler/main.sol]: @shriyatyagii Same here.

File: contracts/mainnet/connectors/euler/main.sol (hunk 2 above)
Comment (shriyatyagii) [reply contracts/mainnet/connectors/euler/main.sol]: Ah! okay

File: contracts/mainnet/connectors/euler/main.sol (hunk 3 above)
Comment (shriyatyagii) [reply contracts/mainnet/connectors/euler/main.sol]: done

**Review by thrilok209 — APPROVED** [review]

**Pull request merged**